odl/contrib/torch/operator.py

```diff
@@ -329,51 +310,39 @@ def forward(self, x):
         >>> odl_op.range.shape
         (2,)
         >>> op_mod = OperatorAsModule(odl_op)
-        >>> t = torch.ones(3)
-        >>> x = autograd.Variable(t[None, :]) # "fake" batch axis
+        >>> x = torch.ones((1, 3))  # with trivial batch axis
         >>> op_mod(x)
-        Variable containing:
-         1  2
-        [torch.FloatTensor of size 1x2]
+        tensor([[1., 2.]])
         >>> t = torch.ones(3)
-        >>> x_tensor = torch.stack([0 * t, 1 * t])
-        >>> x = autograd.Variable(x_tensor)  # batch of 2 inputs
+        >>> x = torch.stack([0 * t, 1 * t])  # batch size 2
         >>> op_mod(x)
-        Variable containing:
-         0  0
-         1  2
-        [torch.FloatTensor of size 2x2]
+        tensor([[0., 0.],
+                [1., 2.]])
 
         An arbitrary number of axes is supported:
 
-        >>> x = autograd.Variable(t[None, None, :])  # "fake" batch and channel
+        >>> x = t[None, None, :]  # trivial batch and channel
         >>> op_mod(x)
-        Variable containing:
-        (0 ,.,.) =
-          1  2
-        [torch.FloatTensor of size 1x1x2]
-        >>> x_tensor = torch.stack([torch.stack([0 * t, 1 * t]),
-        ...                         torch.stack([2 * t, 3 * t]),
-        ...                         torch.stack([4 * t, 5 * t])])
-        >>> x = autograd.Variable(x_tensor)  # batch of 3x2 inputs
+        tensor([[[1., 2.]]])
+        >>> x = torch.stack([torch.stack([0 * t, 1 * t]),
+        ...                  torch.stack([2 * t, 3 * t]),
+        ...                  torch.stack([4 * t, 5 * t])])
         >>> op_mod(x)
-        Variable containing:
-        (0 ,.,.) =
-           0   0
-           1   2
+        tensor([[[ 0.,  0.],
+                 [ 1.,  2.]],
         <BLANKLINE>
-        (1 ,.,.) =
-           2   4
-           3   6
+                [[ 2.,  4.],
+                 [ 3.,  6.]],
         <BLANKLINE>
-        (2 ,.,.) =
-           4   8
-           5  10
-        [torch.FloatTensor of size 3x2x2]
+                [[ 4.,  8.],
+                 [ 5., 10.]]])
         """
         in_shape = x.data.shape
-        op_in_shape = self.op_func.operator.domain.shape
-        op_out_shape = self.op_func.operator.range.shape
+        op_in_shape = self.operator.domain.shape
+        if self.operator.is_functional:
+            op_out_shape = ()
```

Comment:
We could give `Field` a shape `()` to avoid these special cases, if we wanted to.

Reply:
Yes, indeed. It's also related to #1435